XWIKI-20696: Accessibility best-practice: use landmarks in view pages

Sereza7 opened the pull request:
**Jira:** https://jira.xwiki.org/browse/XWIKI-20696

**This is the continuation of this PR: https://github.com/xwiki/xwiki-platform/pull/2106 . I recreated it since after some bad git command, the PR was flooded with commits unrelated to this issue. All the comments on the previous pull request have been taken into account here.**

**Forum discussion:** https://forum.xwiki.org/t/accessibility-best-practice-landmarks/12051

**PR Changes:**
* Added header and footer landmarks.
* Added main landmark for view pages.
* Added navigation and complementary landmarks for panels (with aria-labels)
* Added landmarks for contentheader, contentfooter, extradoc (comments, attachment, ...) and breadcrumbs.


***Note:*** The examples under have been taken without the `xwiki-platform-panels-ui` xar updated. That's why all the panels are landmarked as `Complementary`.

**View:**
Landmarks are highlighted with https://addons.mozilla.org/en-US/firefox/addon/landmarks/ .
Dashed boxes are `guessed` landmarks. Those are not read by a default configuration NVDA. However they might show up with some tools. These guessed landmarks are based on `class` (and not `role` or element nature).
e.g. there's three element with the class `nav` in the Application panel and the tool guesses that there are three navs here.
It would be a good practice to avoid using such classes.
See https://github.com/matatk/landmarks/blob/1ef7cbd194ecbfc8516c40961b90965c71dbe2f6/src/assemble/_landmarksFinder.js#L451-L477 for examples of ids/classes to avoid.


*Before*
![19450-individualLabels](https://user-images.githubusercontent.com/28761965/223479870-f0bc3f2d-8577-4dd9-b993-d06fee0b25b2.png)
*After*
![20696-afterInitialCommit](https://user-images.githubusercontent.com/28761965/223480820-b22b4a12-ca3e-4914-a931-772aec9ae06a.png)

**Video demo:**
It's easier to navigate through the page using a screen reader. With NVDA, the D (and shift D) key press sends to the next landmark.

https://user-images.githubusercontent.com/28761965/223481055-e1348b86-dab8-4d82-97bb-1aa3a062303c.mp4

Sereza7 commented:
*PR Changes:*
* Fixed comments 
* Translation key value formatting

Sereza7 commented:
Note: here the `main` landmark is on the `id='mainContentArea'` element

michitux commented on xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/contentheader.vm:
```diff
@@ -18,7 +18,8 @@
 <div class="row document-header">
-  <div class="document-info col-xs-12 #if($displayContentMenu)col-md-7#end">
+  <section class="document-info col-xs-12 #if($displayContentMenu)col-md-7#end"
+    aria-labelledby="document-title">
```
Is it really good to give that section the title of the page as label? The page title is not really a label of the section imho, something like "Page metadata" would be more fitting imho.

Sereza7 replied:
Some info about the accessible name for region elements (default role of HTML of sections):

[region](https://w3c.github.io/aria/#region) Accessible name required 	
    Helps screen reader users understand the context and purpose of the landmark.
    Use aria-labelledby if a visible label is present, otherwise use aria-label.
    See the [Region Landmark](https://www.w3.org/WAI/ARIA/apg/practices/landmark-regions/#aria_lh_region) section.

The title of the page is not enough :+1: 

I'm not sure I like the term 'metadata'. It's quite generic, and we have other sections that could use this name. Moreover, 'metadata' is a technical term and not friendly to non-technical users. 
I propose 'Page header', and 'Page actions' on the nav next to it.

https://trends.google.com/trends/explore?date=today%205-y&q=metadata,header&hl=en

Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/875e1746bcc7cc789df1a47831609ef8632c8ad8

michitux reviewed (commented):
Do I understand this correctly that, e.g., in the page index, there will now be a `nav` element for every line? I'm not sure this is a good idea. Also, I'm wondering why you haven't assigned a navigation role to the applications panel.

michitux commented:
> Do I understand this correctly that, e.g., in the page index, there will now be a `nav` element for every line? I'm not sure this is a good idea. Also, I'm wondering why you haven't assigned a navigation role to the applications panel.

I'm referring to https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Navigation_Role#accessibility_concerns here which says:

> [Landmark roles](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles#3._landmark_roles) are intended to be used sparingly, to identify larger overall sections of the document. Using too many landmark roles can create "noise" in screen readers, making it difficult to understand the overall layout of the page.

So I think it is very bad idea to make every breadcrumb on a page a navigation element. Instead, only the main breadcrumb at the top of the page should have a navigation role imho.

Sereza7 commented on xwiki-platform-core/xwiki-platform-livetable/xwiki-platform-livetable-ui/src/main/resources/XWiki/LiveTableResultsMacros.xml:
```diff
@@ -396,7 +396,14 @@
-      'doc_location': "#hierarchy($docReference, {'limit': 5, 'plain': false, 'local': true, 'displayTitle': false})",
+      'doc_location': "#hierarchy($docReference,
+        {'limit': 5,
+          'label': $services.localization.render('web.hierarchy.label.document')
```
Fixed the 'same label' part in https://github.com/xwiki/xwiki-platform/pull/2133/commits/ddd6b5d48f20ad236a1ee6db84c174546f653cdd :
![20696-locationChange](https://github.com/xwiki/xwiki-platform/assets/28761965/25fab7ff-e1c7-4692-9984-bb42203beac6)

Sereza7 commented:
@michitux 

> Also, I'm wondering why you haven't assigned a navigation role to the applications panel.

Mistake from me being lazy and not updating all the files on my new working distribution.
Here is the current result with all the files updated:
![20696-OverviewIndex](https://github.com/xwiki/xwiki-platform/assets/28761965/74fe01f7-b778-4cf2-bb88-5388f5e392c9)


___
> So I think it is very bad idea to make every breadcrumb on a page a navigation element. Instead, only the main breadcrumb at the top of the page should have a navigation role imho.

I disagree with you, because breadcrumbs are completely in the scope of the navigation element: it contains for the most part a few anchors that redirect to other pages, and is used to navigate through the wiki. 
If it comes to it, I think the bad practice here would be to have more than a couple breadcrumb on a page. I think that having that many links on the same page is bound to reduce usability (and accessibility in the same way). 

I think it's still okay since the Index page is very special (most pages won't hold more than a couple breadcrumbs). Note that this issue also arises for the Solr search page, all results have a breadcrumb within them.
**TODO:** Give individual labels to all the solr search breadcrumbs

michitux commented on xwiki-platform-core/xwiki-platform-oldcore/src/main/resources/ApplicationResources.properties:
```diff
@@ -739,6 +739,9 @@ platform.core.profile.watchlist.unwatch=Remove from my watch list
 core.footer.modification=Last modified by {0} on {1}
+
+core.document.header.info.label=Page header
+core.document.header.menu.nav.label=Page actions
```
I think it would be good to have an explanation for these strings as comments, in particular as these are not normally shown to the user.

Sereza7 replied:
Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/065f12c47602b8e7eca5dd69fec16652290aa38b

michitux reviewed (commented):
> > So I think it is very bad idea to make every breadcrumb on a page a navigation element. Instead, only the main breadcrumb at the top of the page should have a navigation role imho.
> 
> 
> I disagree with you, because breadcrumbs are completely in the scope of the navigation element: it contains for the most part a few anchors that redirect to other pages, and is used to navigate through the wiki.
> If it comes to it, I think the bad practice here would be to have more than a couple breadcrumb on a page. I think that having that many links on the same page is bound to reduce usability (and accessibility in the same way).

I'm not convinced by this argument. Having many links on a page is a common pattern for index pages and I don't think every group of a few of them should be marked as a navigation landmark. I'm not even sure that in the case of the page move or copy feature for example they should be navigation landmarks as they are really not navigation landmarks, they are just a way to display the full name of a page.

I'm not an accessibility expert so I'm not saying that I know what the best solution should be. Maybe you could open a proposal on the forum to get some more opinions? Or we could consult, e.g., https://ux.stackexchange.com/ to get a more expert opinion?

Apart from that, I have some doubts regarding double escaping. Please check all usages of the hierarchy macro where you added a label to see if you're escaping enough or double escaping as this is not consistent from what I could see.

Sereza7 commented on xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-ui/src/main/resources/XWiki/SolrDocumentDoesNotExistUIX.xml:
```diff
@@ -288,7 +288,13 @@
+      &lt;li&gt;&lt;a
+        href="$escapetool.xml($suggestionDocument.getURL())"&gt;
+        #hierarchy($suggestionResultsDocReference, {
+          'label': $services.model.serialize($docReference, 'local'),
```
The others are double escaping, no change needed here :+1:

Sereza7 commented on xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-ui/src/main/resources/Main/SolrSearchMacros.xml:
```diff
@@ -428,7 +428,7 @@
 #macro (displaySearchResult_document $searchResult)
   #displaySearchResultTitle()
-  #displaySearchResultLocation()
+  #displaySearchResultLocation($searchResult $escapetool.xml($searchResult.title_))
```
Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/d23de1647373837a7d522a8fea83f2383bf6a042

Sereza7 commented on xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-ui/src/main/resources/Main/SolrSearchMacros.xml:
```diff
@@ -451,7 +451,7 @@
     &lt;/a&gt;
   &lt;/h2&gt;
-  #displaySearchResultLocation($searchResult)
+  #displaySearchResultLocation($searchResult $escapetool.xml($searchResultReference.name))
```
Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/d23de1647373837a7d522a8fea83f2383bf6a042

Sereza7 commented on xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-ui/src/main/resources/Main/SolrSearchMacros.xml:
```diff
@@ -467,14 +467,14 @@
     $services.icon.renderHTML('cube') $escapetool.xml($searchResult.propertyname)
   &lt;/h2&gt;
-  #displaySearchResultLocation($searchResult)
+  #displaySearchResultLocation($searchResult $escapetool.xml($searchResult.propertyname))
```
Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/d23de1647373837a7d522a8fea83f2383bf6a042

Sereza7 commented on xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-ui/src/main/resources/XWiki/Results.xml:
```diff
@@ -65,7 +65,8 @@
             'excludeSelf': true,
-            'limit': 6
+            'limit': 6,
+            'label': $escapetool.xml($itemDoc.plainTitle)
```
Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/d23de1647373837a7d522a8fea83f2383bf6a042

Sereza7 commented on xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-ui/src/main/resources/Main/SolrSearchMacros.xml:
```diff
@@ -467,14 +467,14 @@
     $escapetool.xml("${searchResult.get('class').get(0)}[$searchResult.number]")
   &lt;/h2&gt;
-  #displaySearchResultLocation($searchResult)
+  #displaySearchResultLocation($searchResult $escapetool.xml("${searchResult.get('class').get(0)}[$searchResult.number]"))
```
Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/d23de1647373837a7d522a8fea83f2383bf6a042

Sereza7 commented on xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/TemplateProviderMacros.xml:
```diff
@@ -65,6 +65,7 @@
             value="$escapetool.xml($path)" /&gt;
         #end
         #set ($spaceReference = $services.model.resolveSpace($path))
+        #set ($discard = $hierarchyOptions.put('label', $escapetool.xml($path)))
```
Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/d1487e81c189f8659c0e33d36327763227bb7b99

Sereza7 commented on xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/TemplateProviderSheet.xml:
```diff
@@ -184,7 +184,8 @@
             'local': true,
-            'selfIsActive': false
+            'selfIsActive': false,
+            'label': $!escapetool.xml($services.localization.render($doc.getPageReference().getName()))
```
Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/d1487e81c189f8659c0e33d36327763227bb7b99

Sereza7 commented on xwiki-platform-core/xwiki-platform-oldcore/src/main/resources/ApplicationResources.properties:
```diff
@@ -1594,6 +1597,7 @@ core.versions.delete.goback=go back
 
 core.pdf.tableOfContents=Table of Contents
 
+panels.label={0} panel
```
Used comment:
`## Label of a panel on the page. The parameter is the title of the panel.`
`## E.g. in English, for the 'Tips' panel, the label would be 'Tips panel'`

Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/d1487e81c189f8659c0e33d36327763227bb7b99

Sereza7 commented:
Started a discussion on https://forum.xwiki.org/t/ui-accesibility-breadcrumbs-in-the-document-index/12866

michitux commented on xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/hierarchy_macros.vm:
```diff
@@ -146,7 +149,17 @@
+  ## Set as a navigation role because it contains mostly links
+  ## Except if the 'navigation' option is set to false, for special cases where a navigation would clutter
+  ## See discussion on https://forum.xwiki.org/t/ui-accesibility-breadcrumbs-in-the-document-index/12866
+  <#if($!options.navigation == false)div#{else}nav#end aria-label=
```
Citing https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label:

> Note: While aria-label is allowed on any element that can have an accessible name, in practice, aria-label is only supported on interactive elements, widgets, landmarks, images and iframes.

It therefore doesn't make any sense to set an `aria-label` when the element is a `div`. Further, it is therefore also not useful to introduce a ton of new translation keys to set labels on elements that cannot be labelled.

I'm wondering if it wouldn't be better to leave the `#hierarchy` macro alone to avoid that some code introduces accidentally a ton of landmarks on a page and to instead wrap the hierarchy at the top of the page in a `<nav>` element. Alternatively, we could say that the parameter for the label is named `navigationLabel` and that the presence of this parameter triggers the `<nav>` element. But still, the wrapping `<div>` is completely useless and any caller that wants the navigation role could also simply wrap the breadcrumbs in an appropriately labelled `<nav>` element.

michitux commented on xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/widgets/locationPicker.js:
```diff
@@ -276,7 +276,7 @@ require(['jquery', 'xwiki-meta', 'xwiki-events-bridge', 'xwiki-form-validation-a
       }, function(data) {
         // Update the space reference part of the new location.
-        var newLocationContainer = $(data);
+        var newLocationContainer = $(data).find("ol.breadcrumb");
```
Is this change still correct for non-navigation location pickers?

Sereza7 replied:
Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/0e679c1789c8626d7a747ef0f835fa35ca978cb7 :+1:

Sereza7 commented on xwiki-platform-core/xwiki-platform-livetable/xwiki-platform-livetable-ui/src/main/resources/XWiki/LiveTableResultsMacros.xml:
```diff
@@ -396,7 +396,13 @@
+          'local': true,
+          'displayTitle': false
+        }
+      )",
```
Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/bec97203e44123973225a8864d4a32490a1321c7

Sereza7 commented on xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-ui/src/main/resources/XWiki/SolrDocumentDoesNotExistUIX.xml:
```diff
@@ -288,7 +288,12 @@
+        #hierarchy($suggestionResultsDocReference, {
+          'plain' : true, 'local' : true
+        })
+      &lt;/a&gt;&lt;/li&gt;
```
:+1: Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/996debcdc2bd7d6b6f9735ca654844de9fc047e8

Sereza7 commented on xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-ui/src/main/resources/Main/SolrSearchMacros.xml:
```diff
@@ -430,7 +430,7 @@
 #macro (displaySearchResult_document $searchResult)
   #displaySearchResultTitle()
-  #displaySearchResultLocation()
+  #displaySearchResultLocation($searchResult)
```
:+1: Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/991fa262f6310e469c49f475ba838b3111acf154

Sereza7 commented on xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-ui/src/main/resources/Main/SolrSpaceFacet.xml:
```diff
@@ -67,6 +67,7 @@
     #end
     #set ($discard = $macro.data.add({
       'reference': $reference,
+      'navigation': true,
```
![20696-searchNav](https://github.com/xwiki/xwiki-platform/assets/28761965/b1319a41-42fb-4d0c-80ab-c62abfcf6e5c)

The main purpose of this hierarchy is to display location, not navigate :+1: 
Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/dbabcfd57b309fcec540b16b7f3e06ccee1b4165

Sereza7 commented on xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/notification/email/macros.vm:
```diff
@@ -92,7 +92,11 @@
-      #hierarchy($event.document, {'id': 'hierarchy', 'limit': 5, 'treeNavigation': false, 'plain': true})
+      #hierarchy($event.document, {
+      'id': 'hierarchy',
+      'navigation': true,
```
I had a difficult time understanding where this template is used.
Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/057c8ed1f3f3bb49a4392ff58282c877c63c6522 :+1:

Sereza7 commented on xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/locationPicker_macros.vm:
```diff
@@ -71,6 +71,12 @@
+          #set($options.label = $titleField.label)
+        #end
+        #if($options.label)
+          #set($options.navigation = true)
```
:+1: Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/e837656b657170fb1981b2ffeeea86f407e0cbb3

Sereza7 commented on xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/locationPicker_macros.vm:
```diff
@@ -71,6 +71,12 @@
       <div class="breadcrumb-container">
         ## Note: We display only the parent reference here. The new document part will be added from JavaScript.
+        #if(!$options.label && $titleField.label)
+          #set($options.label = $titleField.label)
```
Fixed by https://github.com/xwiki/xwiki-platform/pull/2133#discussion_r1328608044 :+1:

Sereza7 commented on xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/hierarchy_macros.vm:
```diff
@@ -329,6 +344,8 @@
     #set ($options = {
       'id': 'hierarchy',
+      'navigation': true,
+      'label': $doc.getPageReference().getName(),
```
This option.label field is not the exact value of the label added on the navigation object:
https://github.com/xwiki/xwiki-platform/blob/8d7ef8dbbe288870810feba3053b46a95ccd9a0b/xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/hierarchy_macros.vm#L157

We can just make the 'label' field empty for the value of the label to be: 'Breadcrumb'
https://github.com/xwiki/xwiki-platform/blob/68dfa4777f30e7f7fea0c65171f3e8d5c54f1a41/xwiki-platform-core/xwiki-platform-oldcore/src/main/resources/ApplicationResources.properties#L4006-L4007

Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/97e65a6aaf5b14eef98783f6768af3b79112a365

Sereza7 commented on xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/macros.vm:
```diff
@@ -819,22 +819,36 @@ $fn
-<div class="#if ($isLarge)large #{end}panel expanded $!specialClassAttribute $!deprecatedClassAttribute">
+<div role="#if ($isNavigation)navigation#{else}complementary#{end}" class="#if ($isLarge)large #{end}panel
+expanded $!specialClassAttribute $!deprecatedClassAttribute"
+aria-label="$escapetool.xml($services.localization.render('panels.label', [$title]))">
```
Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/4fa30ed3d13c1937e80f28f6d92070287b976485 :+1:

Sereza7 commented on xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/widgets/locationPicker.css:
```diff
@@ -3,12 +3,15 @@
      For advanced users the .location-edit element that follows the breadcrumb has a top margin. */
   margin-bottom: 0;
-  /* Keep the same size whatever the content. */
-  flex-grow: 1;
```
Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/0e679c1789c8626d7a747ef0f835fa35ca978cb7 :+1:

Sereza7 commented on xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/hierarchy.vm:
```diff
@@ -32,6 +32,9 @@
-    #hierarchy($NULL {'id': 'hierarchy', 'limit': 5, 'treeNavigation': true})
+    #hierarchy($NULL {'id': 'hierarchy',
+    'navigation': true,
+    'label': $doc.getPageReference().getName(),
```
Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/2763c7af3c83cd0ca2297a779db4d435168ef8ff :+1: 
Since this is the only breadcrumb with the navigation role left, there is no need to find a more precise name than 'Breadcrumb' :)

Sereza7 commented:
Addressed all of the comments above.

## View
![20696-viewPRUpdated18Sept](https://github.com/xwiki/xwiki-platform/assets/28761965/46ebc87f-508f-4cae-a633-c4fc8464ef47)

michitux commented:
That most recent screenshot looks wrong:

1. Shouldn't the navigation at the top have only the name "Breadcrumb" and not "Breadcrumb for Search"?
2. Why are "Navigation" and "Applications" not navigation landmarks? Also, seeing this (we would get "Navigation navigation"), I'm wondering if the "Panel" suffix/prefix for panels wouldn't be a good idea in the end. But not sure, it might be too technical.
3. "Content information" at the bottom seems strange at least for the default content, did you set that label?

Sereza7 commented:
> 1. Shouldn't the navigation at the top have only the name "Breadcrumb" and not "Breadcrumb for Search"?

This was a mistake from my part, not updated the hierarchy.vm file with the latest version on my local distribution. However I found out a mishaps happening with the changes in XWIKI-18838.
Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/68ac2d1b081b1cee2220fc509070fcf8d732ab64 :+1: 

> 2. Why are "Navigation" and "Applications" not navigation landmarks? Also, seeing this (we would get "Navigation navigation"), I'm wondering if the "Panel" suffix/prefix for panels wouldn't be a good idea in the end. But not sure, it might be too technical.
    
I did not update the Panel pages. Applications is not a navigation landmark since it's not a panel in the 'Navigation' category.
I'm not sure the difference with a prefix will have a huge impact. We can still raise it as a new issue if it ends up being not descriptive enough.

> 3. "Content information" at the bottom seems strange at least for the default content, did you set that label?

I did not set that label, this is the tool's way to announce the footer.


___
Here is a view of the page with all the proper changes, and the update to the breadcrumb from https://github.com/xwiki/xwiki-platform/pull/2133/commits/68ac2d1b081b1cee2220fc509070fcf8d732ab64 :
![20696-simpleNavigation](https://github.com/xwiki/xwiki-platform/assets/28761965/d0dd4c16-ceb8-49ae-a40c-d466927de681)

Note that the Navigation panel is just labelled 'Navigation' (ref to 2.). That's not the best, but it's only one panel and it's consistent across pages and with the visual information conveyed.

michitux commented on xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/hierarchy.vm:
```diff
@@ -33,7 +33,8 @@
     ##
-    #hierarchy($NULL {'id': 'hierarchy', 'limit': 5, 'treeNavigation': true})
+    #hierarchy($NULL {'id': 'hierarchy',
+    'limit': 5, 'treeNavigation': true})
```
The additional line break seems to be the only change now, I think it should be removed now.

michitux commented on xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/hierarchy_macros.vm:
```diff
@@ -146,7 +146,8 @@
   #end
-  <ol #if("$!options.id" != '')id="$options.id"#end class="$breadCrumbClassNames" $breadCrumbParameters><li
+  <ol #if("$!options.id" != '')id="$options.id"#end class="$breadCrumbClassNames" $breadCrumbParameters>
+  <li
```
I think this line break should be removed again, too.

Sereza7 replied:
Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/551cce136d80c1ba8703aa789488c1450d0d79cf :+1:

michitux commented on xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/hierarchy.vm:
```diff
@@ -24,7 +24,7 @@
   ##                   CONTROLLER
   ##
-  ## Call the appropiate breadcrumb depending on the 
+  ## Call the appropiate breadcrumb depending on the
```
```suggestion
  ## Call the appropiate breadcrumb depending on the 
```

While the trailing space is wrong, it's the only change so shouldn't be committed here.

Sereza7 replied:
... My IDE was auto removing the trailing line on save of edited lines...
Addressed in https://github.com/xwiki/xwiki-platform/pull/2133/commits/f969fc7ea1210f19c65693bfb0fbd9fdb635097a :+1:

michitux commented on xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/macros.vm:
```diff
@@ -1521,7 +1535,7 @@ $Msz MB##
     #foreach ($column in $columns)
       #set ($columnProperties = $columnsProperties.get($column))
-      ## Note: Even though the column is hidden, we still need a (potentially hidden) TD to generate valid HTML
+      ## Note: Even though the column is hidden, we still need a (pottentially hidden) TD to generate valid HTML
```
```suggestion
      ## Note: Even though the column is hidden, we still need a (potentially hidden) TD to generate valid HTML
```

This weird change seems to introduce a typo.

Sereza7 replied:
I'm pretty sure this is fixing a typo: https://dictionary.cambridge.org/dictionary/english/potentially
I figured that it was okay to leave it in this PR since this is not the only change for this file.

michitux replied:
Your change replaces "potentially" by "pottentially" which introduces a typo unless I misunderstood something. The change I'm suggesting fixes it again. Fixing a typo would be okay I think, but my point here is that it isn't fixing but introducing one.

Sereza7 replied:
Sorry, I misread the changes

michitux commented on xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/hierarchy.vm:
```diff
@@ -24,7 +24,7 @@
   ##                   CONTROLLER
   ##
-  ## Call the appropiate breadcrumb depending on the 
+## Call the appropiate breadcrumb depending on the 
```
```suggestion
  ## Call the appropiate breadcrumb depending on the 
```

Now we have a change in indentation...
